### DISCUSSION TITLE
(Feature Request) JSON to EE Tags

### DISCUSSION THREAD

**Discussion by blairliikala** [ExpressionEngine/ExpressionEngine#2940] [Feature Requests]
Would like to request a more elegant way of going from JSON to EE tags within an add-on.  Currently this needs additional array manipulation to get all the nested objects and arrays in a JSON response.  Given the popularity and how most [dare I say all?] external services use JSON with their APIs, it would be much easier to integrate 3rd party services within an addon.

Preferred:

```
json =
{
  "item" : "data",
  "nested" : {
     "nested1" : "nested data"
  }
  "nested array" : [
     {
         "data" : "more data"
     },
     {
         "data"  : "even more data"
     }
  ]
}
```
Add-on:
`return ee()->TMPL->parse_variables_using_json($tagdata, json_decode($json));`

Template:
```
{exp:my_addon:json}
  {item}
  {nested:nested1}
  {nested_array}
      {nested_array:data}
  {nested_array}
{/exp:my_addon:json}
```

**Comment by robsonsobral** [comment]
Hi! I offered myself to code a plugin to allow the use of [Laravel-Mix](https://laravel-mix.com/) `mix-manifest.json` on templates. However, I guess a generic JSON parser could be more useful.

May I propose a PR on this? What do you think, @ops-andy ? An add-on just like `exp:markdown`?

**Comment by ops-andy** [comment]
@TomJaeger see @robsonsobral 's proposal above.
